### ISSUE TITLE
The severity by Trivy scanner does not match the CVSSv3

### ISSUE THREAD

**Issue by reasonerjt** [goharbor/harbor#14983]
The CVE with score 9.0+ is marked as `Low`.  This may be due to the impl of the scanner, I'm opening the issue for the record.
We may clarify and push Trivy to fix it or introduce more attribute to the vuln to avoid confusion.
See the screenshot 
<img width="1313" alt="Screen Shot 2021-05-26 at 3 40 12 PM" src="https://user-images.githubusercontent.com/2390463/119627670-fdc92600-be3e-11eb-9c2b-ab87c9f4f9bc.png">

**Comment by phin1x** [comment]
@reasonerjt i already opened a issue two months ago: https://github.com/aquasecurity/trivy/issues/879

the reason is ubuntu. they are using the severity a the priority: https://ubuntu.com/blog/securing-open-source-through-cve-prioritisation

**Comment by reasonerjt** [comment]
@phin1x Thanks 
let's have one written up in Harbor's repo as a similar issue may happen to other scanners.

**Comment by dioguerra** [comment]
I got the similar issue for openjdk:8-jre image
![CVE missmatch](https://user-images.githubusercontent.com/5084445/134166667-2cffe795-c39f-426e-970f-1d0c20f3529e.png)

With aquasecurity/trivy#879 i understand what is happening. So this is a trivy-db problem/option.
From the trivy-db it was added in v0.17.0 the support for the new vulnerability API which now accepts (in the json object) a new field with VendorSeverity in parallel with Severity.

Not sure yet what Harbor considers as the field for the presented severity. Can this be computed according to CVSS3? What is the plan here?

**Comment by danielpacak** [comment]
At the time when we integrated Trivy with Harbor it only recognised CVSS v2 schema. Only recently CVSSV3 column was added in Harbor Portal. Even though Trivy returns VendorScore with CVSS v2 and v3 data I don't think we adapter mapping logic in https://github.com/aquasecurity/harbor-scanner-trivy. VendorScore is also relatively new thing data returned by Trivy. I'll revisit the mapping logic from adapter service to Harbor UI and keep you posted in this thread.

**Comment by dioguerra** [comment]
Using trivy in server mode also shows an interesting thing. For this CVE there are 2 CVSS objects but Harbor uses the lowest score. I'm guessing there is a CVSS order problem here? I would expect the score to be the highest of the CVSS objects.

```trivy client --format json tomcat@sha256:ab07a968b6b98c3e8e0d44d93fdb4f26fd6368119701375f2f669a3b396fae9d | less```

```
{
        "VulnerabilityID": "CVE-2017-7246",
        "PkgName": "libpcre3",
        "InstalledVersion": "2:8.39-12",
        "Layer": {
          "DiffID": "sha256:4dd0c5812fd41bd7ed9a37443595a5ae70ef52da044ae8a3ddbf1b58fe126c54"
        },
        "SeveritySource": "nvd",
        "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2017-7246",
        "Title": "pcre: stack-based buffer overflow write in pcre32_copy_substring",
        "Description": "Stack-based buffer overflow in the pcre32_copy_substring function in pcre_get.c in libpcre1 in PCRE 8.40 allows remote attackers to cause a denial of service (WRITE of size 268) or possibly have unspecified other impact via a crafted file.",
        "Severity": "HIGH",
        "CweIDs": [
          "CWE-119"
        ],
        "CVSS": {
          "nvd": {
            "V2Vector": "AV:N/AC:M/Au:N/C:P/I:P/A:P",
            "V3Vector": "CVSS:3.0/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
            "V2Score": 6.8,
            "V3Score": 7.8
          },
          "redhat": {
            "V3Vector": "CVSS:3.0/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:L",
            "V3Score": 3.7
          }
        },
        "References": [
          "http://www.securityfocus.com/bid/97067",
          "https://access.redhat.com/errata/RHSA-2018:2486",
          "https://blogs.gentoo.org/ago/2017/03/20/libpcre-two-stack-based-buffer-overflow-write-in-pcre32_copy_substring-pcre_get-c/",
          "https://security.gentoo.org/glsa/201710-25"
        ],
        "PublishedDate": "2017-03-23T21:59:00Z",
        "LastModifiedDate": "2018-08-17T10:29:00Z"
      },
```
AND

```
{
        "VulnerabilityID": "CVE-2021-33574",
        "PkgName": "libc-bin",
        "InstalledVersion": "2.28-10",
        "Layer": {
          "DiffID": "sha256:4dd0c5812fd41bd7ed9a37443595a5ae70ef52da044ae8a3ddbf1b58fe126c54"
        },
        "SeveritySource": "nvd",
        "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2021-33574",
        "Title": "glibc: mq_notify does not handle separately allocated thread attributes",
        "Description": "The mq_notify function in the GNU C Library (aka glibc) versions 2.32 and 2.33 has a use-after-free. It may use the notification thread attributes object (passed through its struct sigevent parameter) after it has been freed by the caller, leading to a denial of service (application crash) or possibly unspecified other impact.",
        "Severity": "CRITICAL",
        "CweIDs": [
          "CWE-416"
        ],
        "CVSS": {
          "nvd": {
            "V2Vector": "AV:N/AC:L/Au:N/C:P/I:P/A:P",
            "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H",
            "V2Score": 7.5,
            "V3Score": 9.8
          },
          "redhat": {
            "V3Vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:N/A:H",
            "V3Score": 5.9
          }
        },
        "References": [
          "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/RBUUWUGXVILQXVWEOU7N42ICHPJNAEUP/",
          "https://security.gentoo.org/glsa/202107-07",
          "https://security.netapp.com/advisory/ntap-20210629-0005/",
          "https://sourceware.org/bugzilla/show_bug.cgi?id=27896",
          "https://sourceware.org/bugzilla/show_bug.cgi?id=27896#c1"
        ],
        "PublishedDate": "2021-05-25T22:15:00Z",
        "LastModifiedDate": "2021-07-07T03:15:00Z"
      },
```

**Comment by dioguerra** [comment]
@danielpacak I did a check again, and this seems to be solved for harbor v2.3.2.
Forgive my noise

**Comment by oDn** [comment]
Unless I'm mistaken, this seems to be fixed in v2.3.2.

However, the value of the CVSS v3 score still doesn't reflect what is published. As an example, see [CVE-2021-3711](https://nvd.nist.gov/vuln/detail/CVE-2021-3711). 
Harbor/Trivy reports a score of 8.1 whereas NIST reports a Base Score of 9.8.

This seems to be the same issue reported by @dioguerra  just [above](https://github.com/goharbor/harbor/issues/14983#issuecomment-924034676).

**Comment by dioguerra** [comment]
Yes, the issues are mostly resolved in Harbor v2.3.2 but there are still inconsistencies on the Severity and the Score.
I noticed that Severity has the maximum value of the CVSS scores available and most of the times, the score presented on Harbor is the last one in the CVSS array (usually redhat score takes priority over nvd).

**Comment by danielpacak** [comment]
For the update, I'm still looking into this issue but first we have to fix #15623 , which IMO may introduce data consistency errors.
